components/codec/src/byte.rs

```diff
@@ -869,6 +942,97 @@ mod tests {
         }
     }
 
+    #[test]
+    fn test_memcmp_encode_all_in_place() {
+        // Checks whether encoded result matches expectation.
```

Comment:
I think you can simply embed your test with current encode tests (verify that your in place encoding produce the same result as the non-inplace one).